pkg/vault/contracts/RouterCommon.sol

```diff
@@ -263,12 +263,16 @@ abstract contract RouterCommon is IRouterCommon, VaultGuard {
     function _sendTokenOut(address sender, IERC20 tokenOut, uint256 amountOut, bool wethIsEth) internal {
         // If the tokenOut is ETH, then unwrap `amountOut` into ETH.
         if (wethIsEth && tokenOut == _weth) {
-            // Receive the WETH amountOut.
+            // Receive the WETH amountOut. We want to call `sendTo` regardless for reentrancy protection;
+            // it has an internal check for zero amounts.
```

Comment:
Hmm could you explain why this is an issue? A zero value `sendTo` has no effect anyways, as it shouldn't modify the state at all.

Reply:
maybe just return early if `amountOut` is 0?

Reply:
`sendTo` does a transfer; if 0, it would revert on tokens that don't allow it.

It breaks a non-reentrancy test, but I can try to avoid that another way I guess.

Reply:
I have to be careful to always call settle, as that updates reserves and handles DoS dust, etc. I can short-circuit other places to avoid calling sendTo.

Reply:
```
    function settle(IERC20 token, uint256 amountHint) external nonReentrant onlyWhenUnlocked returns (uint256 credit) {
        uint256 reservesBefore = _reservesOf[token];
        uint256 currentReserves = token.balanceOf(address(this));
        _reservesOf[token] = currentReserves;
        credit = currentReserves - reservesBefore;

        // If the given hint is equal or greater to the reserve difference, we just take the actual reserve difference
        // as the paid amount; the actual balance of the tokens in the vault is what matters here.
        if (credit > amountHint) {
            // If the difference in reserves is higher than the amount claimed to be paid by the caller, there was some
            // leftover that had been sent to the vault beforehand, which was not incorporated into the reserves.
            // In that case, we simply discard the leftover by considering the given hint as the amount paid.
            // In turn, this gives the caller credit for the given amount hint, which is what the caller is expecting.
            credit = amountHint;
        }

        _supplyCredit(token, credit);
    }
```

I don't think calling settle with 0 makes any difference. If anything, it will just add the dust to the reserves, which is irrelevant to the current operation.
- If credit is 0, it will do `_supplyCredit(token, 0)`. `_supplyCredit` calls `_accountDelta`, which returns early for a `delta == 0` (i.e. does nothing).
- If credit is > 0, `credit > amountHint --> true`. So the `amountHint` will be assigned to the credit, and then `_supplyCredit(token, 0)` will be called.

In the latter case the reserves are updated, incorporating the dust to the reserves, but that can also be done on the next operation. It shouldn't affect debit / credit settlement nor prevent this operation from finishing.

Reply:
In other words, I think breaking loops early is just fine; we don't need to call `settle` with 0 amounts.